STX.REST.RESTFulSense.Clients/Services/Foundations/ErrorMappers/ErrorMapperService.cs

```diff
@@ -21,7 +21,9 @@ internal partial class ErrorMapperService : IErrorMapperService
         {
             ValidateStatusCode(statusCode);
 
-            StatusDetail statusDetail = errorBroker.SelectAllStatusDetails()
+            IQueryable<StatusDetail> statusDetails = await errorBroker.SelectAllStatusDetailsAsync();
+
+            StatusDetail statusDetail = statusDetails
                 .FirstOrDefault(statusDetail =>
                     statusDetail != null && statusDetail.Code == statusCode);
 
```

Comment:
return await ValueTask.FromResult(statusDetail);

could be simplified with

return statusDetail;

because you just have a previous await done.

Reply:
same as what @hassanhabib said